linera-client/src/client_context.rs

```diff
@@ -41,28 +41,47 @@ use {
     linera_base::{
         crypto::PublicKey,
         data_types::Amount,
+        hashed::Hashed,
         identifiers::{AccountOwner, ApplicationId, Owner},
     },
-    linera_chain::data_types::{
-        BlockProposal, ExecutedBlock, ProposedBlock, SignatureAggregator, Vote,
+    linera_chain::{
+        data_types::{BlockProposal, ExecutedBlock, ProposedBlock, SignatureAggregator, Vote},
+        types::{CertificateValue, ConfirmedBlock, GenericCertificate},
+    },
+    linera_core::{
+        data_types::{ChainInfoQuery, ChainInfoResponse},
+        node::{ValidatorNode, ValidatorNodeProvider},
     },
-    linera_chain::types::{CertificateValue, GenericCertificate},
-    linera_core::data_types::ChainInfoQuery,
     linera_execution::{
         committee::Epoch,
         system::{OpenChainConfig, Recipient, SystemOperation, OPEN_CHAIN_MESSAGE_INDEX},
         Operation,
     },
     linera_rpc::{
         config::NetworkProtocol, grpc::GrpcClient, mass_client::MassClient,
-        simple::SimpleMassClient, RpcMessage,
+        simple::SimpleMassClient, HandleConfirmedCertificateRequest, RpcMessage,
     },
     linera_sdk::abis::fungible,
     std::{collections::HashMap, iter},
     tokio::task,
-    tracing::{error, trace},
+    tracing::{error, trace, warn},
 };
 
+#[cfg(feature = "benchmark")]
+fn deserialize_response(response: RpcMessage) -> Option<ChainInfoResponse> {
```

Comment:
(You could make it a static method of `ChainClient` below; one less `cfg` attribute.)